src/file_watcher.rs

```diff
@@ -0,0 +1,1426 @@
+use crate::{
+    daemonise, enforce_singleton, into_recoverable, telemetry_config, EnvSetting, Result,
+    TelemetryError, WatcherResult,
+};
+
+use base64::{engine::general_purpose::STANDARD, DecodeError, Engine};
+use chrono::NaiveDateTime;
+use influxdb_line_protocol::LineProtocolBuilder;
+use nix::{
+    fcntl::{Flock, FlockArg},
+    sys::signal::{kill, Signal::SIGKILL},
+    unistd::{getpid, Pid},
+};
+use regex::{Captures, Regex};
+use reqwest::blocking::{Client, Request, RequestBuilder};
+use std::fs::ReadDir;
+use std::{
+    clone::Clone,
+    collections::HashMap,
+    env::var,
+    fs::{read_dir, remove_file, File, OpenOptions},
+    io::{BufRead, BufReader, Read},
+    path::{Path, PathBuf},
+    process::exit,
+    sync::{
+        atomic::{AtomicBool, AtomicI32, Ordering},
+        LazyLock,
+    },
+    thread::sleep,
+    time::Duration,
+};
+
+//
+// Module config
+//
+
+/// Configuration for `FileWatcher`
+#[derive(Debug, Clone)]
+struct FileWatcherConfig {
+    // The path to its lockfile
+    lockfile: PathBuf,
+    // The path to its logfile
+    logfile: PathBuf,
+    // The token to use to authenticate with InfluxDB
+    influxdb_token: String,
+    // The URL of the InfluxDB instance
+    influxdb_url: String,
+    // The interval to poll for aged-out telemetry files
+    poll_interval: Duration,
+}
+
+/// Get the `FileWatcher` configuration
+///
+/// This function returns the `'static` `FileWatcher` configuration.
+fn config() -> Result<&'static FileWatcherConfig> {
+    static FILEWATCHER_CONFIG: LazyLock<Result<FileWatcherConfig>> = LazyLock::new(|| {
+        let get_env = |key, default| EnvSetting::new(key, default).get();
+
+        // Since we use an hourly appender, we default to polling every hour
+        let poll_interval = get_env("FILEWATCHER_POLL_INTERVAL", "3600").parse()?;
+
+        // Format the InfluxDB URL (the org name needs to be URL-encoded)
+        let influxdb_url = format!(
+            "{}/api/v2/write?org={}&bucket={}&precision=ns",
+            get_env(
+                "INFLUXDB_URL",
+                "https://us-east-1-1.aws.cloud2.influxdata.com"
+            ),
+            get_env("INFLUXDB_ORG", "Dev%20Team"),
+            get_env("INFLUXDB_BUCKET", "telemetry"),
+        );
```

Comment:
maybe a little easier to read? 

```suggestion
let base_url = get_env("INFLUXDB_URL", "https://us-east-1-1.aws.cloud2.influxdata.com");
let org = get_env("INFLUXDB_ORG", "Dev%20Team");
let bucket = get_env("INFLUXDB_BUCKET", "telemetry");
let influxdb_url = format!("{base_url}/api/v2/write?org={org}&bucket={bucket}&precision=ns");
```

Reply:
Probably not needed anymore since removing magic strings. It looks like this now:

```
 let influxdb_url = format!(
            "{}/api/v2/write?org={}&bucket={}&precision=ns",
            get_env("INFLUXDB_URL", INFLUXDB_URL),
            get_env("INFLUXDB_ORG", INFLUXDB_ORG),
            get_env("INFLUXDB_BUCKET", INFLUXDB_BUCKET),
        );
```